File: go/adapter/ThirdPartyWeatherAPI.go
package main

// Implementación concreta de la API externa
type ThirdPartyWeatherAPI struct{}

func (t ThirdPartyWeatherAPI) GetTempC() float64 {
	return 25
}

func (t ThirdPartyWeatherAPI) GetHumidity() float64 {
	return 60
}

func (t ThirdPartyWeatherAPI) GetWindSpeedKPH() float64 {
	return 15
}

File: ts/adapter/ThirdPartyWeatherAPI.ts
import { WeatherAPI } from "./types";

export class ThirdPartyWeatherAPI implements WeatherAPI {
    getTempC(): number {
        // Simulate fetching temperature in Celsius from a third-party API
        return 25; // Example temperature
    }

    getHumidity(): number {
        // Simulate fetching humidity from a third-party API
        return 60; // Example humidity
    }

    getWindSpeedKPH(): number {
        // Simulate fetching wind speed in KPH from a third-party API
        return 15; // Example wind speed
    }
}